.github/workflows/release.yml

```
@@ -0,0 +1,28 @@
+name: Release
+on:
+  push:
+    tags:
+      - v**
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+          cache: 'gradle'
+
+      - name: release
+        env:
+          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+        run: |
+          # todo: verify version is same as tag
```

Maybe too much of a hack??
```suggestion
      - id: getTag
        name: Get Tag
        run: echo ::set-output name=tag::${GITHUB_REF#refs/*/}
      - name: release
        env:
          SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
          SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
          TAG_NAME: ${{ steps.getTag.outputs.tag }}
        run: |
          if [[ -z $(grep CURRENT_GRPC_KOTLIN_VERSION build.gradle.kts | grep ${TAG_NAME:1}) ]];
          then
            exit 1;
          fi
```

---

I was thinking about something like that. But a few things that we could do:
1. There is now a better way to get the tag: `GITHUB_REF_NAME`
1. I think instead of parsing the build file it might be nice to have a task to get the value of `version` so we don't have to do mucky parsing

Let's keep that for a follow-on task and see if this thing even works first.